packages/statemanager/src/cache/account.ts

```diff
@@ -172,7 +172,7 @@ export class AccountCache extends Cache {
     for (const entry of diffMap.entries()) {
       const addressHex = entry[0]
       const elem = entry[1]
-      if (elem === undefined) {
+      if (elem?.accountRLP === undefined) {
```

Comment:
So the wrong fix from above might actually work when here all the `elem?.accountRLP` entries are deleted.

This has a bad side effect on the cache though since it deletes for all accounts where `elem?.accountRLP === undefined` the information from the cache that this account does not exist in the state (which is *also* a state information).